src/staticanalysis/bot/static_analysis_bot/clang/format.py

```diff
@@ -35,19 +35,32 @@ def __init__(self):
     @stats.api.timed('runtime.clang-format')
     def run(self, revision):
         '''
-        Run ./mach clang-format on the current patch
+        Run ./mach clang-format on all of the C/C++ files from the patch
         '''
         assert isinstance(revision, Revision)
 
-        # Commit the current revision for `./mach clang-format` to reformat its changes
         cmd = [
             'gecko-env',
-            './mach', '--log-no-times', 'clang-format',
+            './mach', '--log-no-times', 'clang-format', '-p'
         ]
-        logger.info('Running ./mach clang-format', cmd=' '.join(cmd))
 
-        # Run command
-        clang_output = subprocess.check_output(cmd, cwd=settings.repo_dir).decode('utf-8')
+        # Append to the cmd list each C/C++ file for format
+        found_eligible_files = False
```

Comment:
what about moving that in a function instead?

Reply:
Sure I have no objections against that.

Reply:
Updated with latest modifications that adds a function which returns a list of files to be sent to format.